# Primer avance PIA
Equipo: 6

Grupo: 003, Lunes de 6:00 a 9:00 pm

Integrantes:
* 1728247 Castillo Cerda Manuel Orlando
* 1854568 Cedillo Hernandez Vanessa Nahomy
* 1847759 Lozano Rangel Antonio de Jesus
* 1795359 Vega Flores Blanca Janeth


## Base de datos principal
1. Título de la base de datos.
  * Nombre: Donación de Órganos en México.
  * Url: https://www.kaggle.com/joshdoit/master
  
  
2. Descripción de los datos.
  * ¿Qué tipo de datos son?  
    Es una tabla que contiene registros sobre el proceso de la donación de órganos.
  * Descripción de las columnas
    * SEXO: Texto corto. Muestra el sexo del donante, el cual puede ser femenino o masculino.
    * CODIGO_SEXO: Valor numérico entero, 1 se refiere a masculino y 2 a femenino.
    * TIPO_DONANTE: Texto corto. Muestra si es un donante vivo o muerto.
    * MUERTE: Texto corto. Muestra la causa de muerte, si no aplica es porque el paciente es de tipo vivo.
    * ENTIDAD_FEDERATIVA: Texto corto. Muestra el nombre de la entidad federativa donde se realiza el procedimiento médico.
    * CODIGO_ENTIDAD_FEDERATIVA: Valor numérico entero. Aquí se asigna un número a cada entidad federativa.
    * ESTABLECIMIENTO: Texto largo. Muestra el nombre del establecimiento donde se realiza el procedimiento médico.
    * INSTITUCION: Texto corto. Nombre de las instituciones que realizan el procedimiento de extracción de órganos
    * EDAD_ANIOS: Valor numérico entero. Muestra la edad del donante.
    * FECHA_PROCURACION: Tiempo (DataTime). Muestra la fecha en la que se realiza el procedimiento médico.
    * RINON_IZQUIERDO: Valor numérico entero. Cantidad de riñones izquierdos donados.
    * RINON_DERECHO: Valor numérico entero. Cantidad de riñones derechos donados.
    * RINON_BLOCK: Valor numérico entero. Cantidad de bloques de riñones donados.
    * PULMON_IZQUIERDO: Valor numérico entero. Cantidad de pulmones izquierdos donados.
    * PULMON_DERECHO: Valor numérico entero. Cantidad de pulmones derechos donados.
    * CORAZON: Valor numérico entero. Cantidad de corazones donados.
    * HIGADO: Valor numérico entero. Cantidad de hígados donados.
    * PANCREAS: Valor numérico entero. Cantidad de páncreas donados.
    * INTESTINO: Valor numérico entero. Cantidad de intestinos donados.
    * CORNEA_IZQUIERDA: Valor numérico entero. Cantidad de corneas izquierdas donadas.
    * CORNEA_DERECHA: Valor numérico entero. Cantidad de corneas derechas donadas.
    * PIEL: Valor numérico entero. Cantidad de veces que ha donado piel.
    * HUESOS: Valor numérico entero. Cantidad de huesos donados.
    * CORAZON_TEJIDOS: Valor numérico entero. Cantidad de tejidos de corazón donados.

  

3. Justificación del uso de datos.  
  * ¿Cuáles son las características que les llamó la atención de los datos?¿Qué les hizo querer trabajar con ellos?  
    Principalmente, que la base de datos está bastante completa, al menos para nuestro objetivo, así como hacer énfasis en que la donación de órganos nos parece un tema del cual se debería conocer más, ya que es algo importante y, sin embargo, mucha gente no cuenta con la cultura de donar. 
  * ¿Qué beneficio encuentran de trabajar con estos datos?  
    Conocer realmente que órganos se requieren más en los hospitales de nuestro país y ver si se cuenta con la accesibilidad a ellos.
  

4. Objetivo final.
  * Objetivo principal.
    * Pronosticar la cantidad de cada órgano que se donara en cada entidad federativa en los próximos años.  
      Utilizaremos las herramientas que estén a nuestro alcance para analizar los datos que nos proporciona nuestra base, la cual comprende un periodo del 2007 al 2019, dividiendo estos por entidad federativa, y ver en cada una si existe cierto patrón que se repita o si hay alguna tendencia en nuestros datos, y de esta manera definir como se comportaran las donaciones de órganos en los próximos años. 
  * Objetivos secundarios.
    * Predecir que órganos se donaran en cada entidad federativa en los próximos años.
      Básicamente, se estaría realizando lo mismo que en objetivo principal, pero en caso de no poder definir una cantidad aproximada de cada órgano, al menos esperamos determinar cuáles de los 14 órganos dados serán donados en cada entidad. 
    * Verificar si con el paso de los años ha incrementado la cultura de donación.
      La idea es dividir nuestra base en 2 grupos, vivos y muertos, y analizar los datos para determinar si existe una tendencia al alza en el grupo de vivos, y una tendencia a la baja en el grupo de muertos, si ambas tendencias se hallan se estaría demostrando que esta incrementado la cultura de donación.


5. Mostrar preguntas de interés.
  *  ¿En que rango de edad se encuentra la mayoría de los donadores?
  *  ¿Qué órgano suele ser el más donado?
  *  ¿Cuál es la institución que recibe más donaciones?
  *  ¿Qué entidad federativa hace más donaciones de órganos?


6. Parte de programación.

#### Importar librerías y base de datos.

In [1]:
import pandas as pd
import numpy as np

# Lectura csv
df_org = pd.read_csv("/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/Donacion organos Mex.csv")

#### Visualizar los datos.  
Primero, deseamos conocer la cantidad de columnas y filas que tiene nuestra base.

In [2]:
df_org.shape

(44583, 24)

Esto indica, que se cuenta con 44,583 filas y 24 columnas.  
Ahora, necesitamos saber cuáles columnas tenemos

In [20]:
df_org.columns

Index(['SEXO', 'CODIGO_SEXO', 'TIPO_DONANTE', 'MUERTE', 'ENTIDAD_FEDERATIVA',
       'CODIGO_ENTIDAD_FEDERATIVA', 'ESTABLECIMIENTO', 'INSTITUCION',
       'EDAD_ANIOS', 'FECHA_PROCURACION', 'RINON_IZQUIERDO', 'RINON_DERECHO',
       'RINON_BLOCK', 'PULMON_IZQUIERDO', 'PULMON_DERECHO', 'CORAZON',
       'HIGADO', 'PANCREAS', 'INTESTINO', 'CORNEA_IZQUIERDA', 'CORNEA_DERECHA',
       'PIEL', 'HUESOS', 'CORAZON_TEJIDOS'],
      dtype='object')

Y posteriormente, hay que conocer los elementos que se tienen en dichas columnas:
  * Primeros 10 elementos de cada una.

In [11]:
df_org.head(10)

SEXO  CODIGO_SEXO TIPO_DONANTE                   MUERTE  \
0   FEMENINO          2.0      CADÁVER  PARO CARDIORESPIRATORIO   
1  MASCULINO          1.0      CADÁVER  PARO CARDIORESPIRATORIO   
2   FEMENINO          2.0      CADÁVER        MUERTE ENCEFÁLICA   
3   FEMENINO          2.0      CADÁVER  PARO CARDIORESPIRATORIO   
4  MASCULINO          1.0      CADÁVER  PARO CARDIORESPIRATORIO   
5   FEMENINO          2.0      CADÁVER        MUERTE ENCEFÁLICA   
6  MASCULINO          1.0      CADÁVER        MUERTE ENCEFÁLICA   
7   FEMENINO          2.0      CADÁVER        MUERTE ENCEFÁLICA   
8   FEMENINO          2.0      CADÁVER  PARO CARDIORESPIRATORIO   
9  MASCULINO          1.0      CADÁVER        MUERTE ENCEFÁLICA   

  ENTIDAD_FEDERATIVA  CODIGO_ENTIDAD_FEDERATIVA  \
0         NUEVO LEÓN                         19   
1   CIUDAD DE MEXICO                          9   
2             PUEBLA                         21   
3          QUERÉTARO                         22   
4             MÉXICO                         15   
5    SAN LUIS POTOSÍ                         24   
6         GUANAJUATO                         11   
7            TABASCO                         27   
8             MÉXICO                         15   
9    SAN LUIS POTOSÍ                         24   

                                     ESTABLECIMIENTO INSTITUCION  EDAD_ANIOS  \
0         U.M.A.E. HOSPITAL DE ESPECIALIDADES No. 25        IMSS        49.0   
1  U.M.A.E. HOSPITAL DE CARDIOLOGIA DEL CENTRO ME...        IMSS        72.0   
2  U.M.A.E. HOSPITAL DE ESPECIALIDADES C.M.N. GRA...        IMSS        61.0   
3                      HOSPITAL GENERAL DE QUERETARO         SSE        71.0   
4  CENTRO MEDICO "LIC. ADOLFO LOPEZ MATEOS" DE TE...         SSE        63.0   
5           HOSPITAL LOMAS DE SAN LUIS INTERNACIONAL     PRIVADO        41.0   
6                         HOSPITAL GENERAL DE CELAYA         SSE        58.0   
7          HOSPITAL GENERAL DE ZONA C/U M.A.  No. 46        IMSS        31.0   
8        HOSPITAL GENERAL DE ECATEPEC "LAS AMERICAS"         SSE        60.0   
9      HOSPITAL CENTRAL "DR. IGNACIO MORONES PRIETO"         SSE        45.0   

  FECHA_PROCURACION  ...  PULMON_DERECHO  CORAZON  HIGADO  PANCREAS  \
0        2007-06-11  ...               0        0       0         0   
1        2008-05-08  ...               0        0       0         0   
2        2009-04-21  ...               0        0       0         0   
3        2010-04-29  ...               0        0       0         0   
4        2011-09-16  ...               0        0       0         0   
5        2012-03-04  ...               0        0       0         0   
6        2013-07-12  ...               0        0       0         0   
7        2014-08-27  ...               0        0       0         0   
8        2015-05-20  ...               0        0       0         0   
9        2016-03-21  ...               0        0       0         0   

   INTESTINO  CORNEA_IZQUIERDA  CORNEA_DERECHA  PIEL  HUESOS  CORAZON_TEJIDOS  
0          0                 1               1     0       0                0  
1          0                 1               1     0       0                0  
2          0                 1               1     0       0                0  
3          0                 0               0     0       1                0  
4          0                 1               1     0       0                0  
5          0                 1               1     0       0                0  
6          0                 0               0     0       1                0  
7          0                 1               1     0       0                0  
8          0                 1               1     1       1                0  
9          0                 1               1     0       0                0  

[10 rows x 24 columns]

  * Últimos 10 elementos de cada una.

In [12]:
df_org.tail(10)

SEXO  CODIGO_SEXO TIPO_DONANTE     MUERTE ENTIDAD_FEDERATIVA  \
44573   FEMENINO          2.0         VIVO  NO APLICA    SAN LUIS POTOSÍ   
44574  MASCULINO          1.0         VIVO  NO APLICA         NUEVO LEÓN   
44575  MASCULINO          1.0         VIVO  NO APLICA     AGUASCALIENTES   
44576   FEMENINO          2.0         VIVO  NO APLICA          CHIHUAHUA   
44577   FEMENINO          2.0         VIVO  NO APLICA    SAN LUIS POTOSÍ   
44578   FEMENINO          2.0         VIVO  NO APLICA   CIUDAD DE MEXICO   
44579   FEMENINO          2.0         VIVO  NO APLICA           COAHUILA   
44580  MASCULINO          1.0         VIVO  NO APLICA   CIUDAD DE MEXICO   
44581  MASCULINO          1.0         VIVO  NO APLICA            JALISCO   
44582   FEMENINO          2.0         VIVO  NO APLICA            SINALOA   

       CODIGO_ENTIDAD_FEDERATIVA  \
44573                         24   
44574                         19   
44575                          1   
44576                          8   
44577                         24   
44578                          9   
44579                          5   
44580                          9   
44581                         14   
44582                         25   

                                         ESTABLECIMIENTO INSTITUCION  \
44573  STAR MEDICA, S. A. DE C. V. (HOSPITAL SAN LUIS...     PRIVADO   
44574       FUNDACION SANTOS Y DE LA GARZA EVIA, I.B.P.      PRIVADO   
44575                 CENTENARIO HOSPITAL MIGUEL HIDALGO         SSE   
44576     SERVICIOS HOSPITALARIOS DE MEXICO S.A. DE C.V.     PRIVADO   
44577                    HOSPITAL GENERAL DE ZONA No. 50        IMSS   
44578  UNIDAD MEDICA DE ALTA ESPECIALIDAD DOCTOR GAUD...        IMSS   
44579         U.M.A.E. HOSPITAL DE ESPECIALIDADES NO. 71        IMSS   
44580                           HOSPITAL CENTRAL MILITAR      SEDENA   
44581                       HOSPITAL DE PEDIATRÍA C.M.O.        IMSS   
44582                      HOSPITAL FATIMA, S.A. DE C.V.     PRIVADO   

       EDAD_ANIOS FECHA_PROCURACION  ...  PULMON_DERECHO  CORAZON  HIGADO  \
44573        39.0        2019-06-21  ...               0        0       0   
44574        54.0        2019-06-26  ...               0        0       0   
44575        39.0        2019-06-27  ...               0        0       0   
44576        60.0        2019-06-26  ...               0        0       0   
44577        21.0        2019-06-27  ...               0        0       0   
44578        44.0        2019-06-06  ...               0        0       0   
44579        33.0        2019-06-27  ...               0        0       0   
44580        43.0        2019-06-27  ...               0        0       0   
44581        21.0        2019-06-25  ...               0        0       1   
44582        45.0        2019-06-29  ...               0        0       0   

       PANCREAS  INTESTINO  CORNEA_IZQUIERDA  CORNEA_DERECHA  PIEL  HUESOS  \
44573         0          0                 0               0     0       0   
44574         0          0                 0               0     0       0   
44575         0          0                 0               0     0       0   
44576         0          0                 0               0     0       0   
44577         0          0                 0               0     0       0   
44578         0          0                 0               0     0       0   
44579         0          0                 0               0     0       0   
44580         0          0                 0               0     0       0   
44581         0          0                 0               0     0       0   
44582         0          0                 0               0     0       0   

       CORAZON_TEJIDOS  
44573                0  
44574                0  
44575                0  
44576                0  
44577                0  
44578                0  
44579                0  
44580                0  
44581                0  
44582                0  

[1

#### Descripción básica de columnas.
Así mismo, es importante saber el tipo de datos de todas las columnas

In [14]:
df_org.dtypes

SEXO                          object
CODIGO_SEXO                  float64
TIPO_DONANTE                  object
MUERTE                        object
ENTIDAD_FEDERATIVA            object
CODIGO_ENTIDAD_FEDERATIVA      int64
ESTABLECIMIENTO               object
INSTITUCION                   object
EDAD_ANIOS                   float64
FECHA_PROCURACION             object
RINON_IZQUIERDO                int64
RINON_DERECHO                  int64
RINON_BLOCK                    int64
PULMON_IZQUIERDO               int64
PULMON_DERECHO                 int64
CORAZON                        int64
HIGADO                         int64
PANCREAS                       int64
INTESTINO                      int64
CORNEA_IZQUIERDA               int64
CORNEA_DERECHA                 int64
PIEL                           int64
HUESOS                         int64
CORAZON_TEJIDOS                int64
dtype: object

Y verificar si nuestra base cuenta con valores nulos

In [15]:
df_org.isnull()

SEXO  CODIGO_SEXO  TIPO_DONANTE  MUERTE  ENTIDAD_FEDERATIVA  \
0      False        False         False   False               False   
1      False        False         False   False               False   
2      False        False         False   False               False   
3      False        False         False   False               False   
4      False        False         False   False               False   
...      ...          ...           ...     ...                 ...   
44578  False        False         False   False               False   
44579  False        False         False   False               False   
44580  False        False         False   False               False   
44581  False        False         False   False               False   
44582  False        False         False   False               False   

       CODIGO_ENTIDAD_FEDERATIVA  ESTABLECIMIENTO  INSTITUCION  EDAD_ANIOS  \
0                          False            False        False       False   
1                          False            False        False       False   
2                          False            False        False       False   
3                          False            False        False       False   
4                          False            False        False       False   
...                          ...              ...          ...         ...   
44578                      False            False        False       False   
44579                      False            False        False       False   
44580                      False            False        False       False   
44581                      False            False        False       False   
44582                      False            False        False       False   

       FECHA_PROCURACION  ...  PULMON_DERECHO  CORAZON  HIGADO  PANCREAS  \
0                  False  ...           False    False   False     False   
1                  False  ...           False    False   False     False   
2                  False  ...           False    False   False     False   
3                  False  ...           False    False   False     False   
4                  False  ...           False    False   False     False   
...                  ...  ...             ...      ...     ...       ...   
44578              False  ...           False    False   False     False   
44579              False  ...           False    False   False     False   
44580              False  ...           False    False   False     False   
44581              False  ...           False    False   False     False   
44582              False  ...           False    False   False     False   

       INTESTINO  CORNEA_IZQUIERDA  CORNEA_DERECHA   PIEL  HUESOS  \
0          False             False           False  False   False   
1          False             False           False  False   False   
2          False             False           False  False   False   
3          False             False           False  False   False   
4          False             False           False  False   False   
...          ...               ...             ...    ...     ...   
44578      False             False           False  False   False   
44579      False             False           False  False   False   
44580      False             False           False  False   False   
44581      False             False           False  False   False   
44582      False             False           False  False   False   

       CORAZON_TEJIDOS  
0                False  
1                False  
2                False  
3                False  
4                False  
...                ...  
44578            False  
44579            False  
44580            False  
44581            False  
44582            False  

[44583 rows x 24 columns]

Dado que se cuenta con bastantes celdas, es más sencillo mostrar en que columnas se cuenta con dichos valores

In [17]:
df_org.isnull().any()

SEXO                          True
CODIGO_SEXO                   True
TIPO_DONANTE                 False
MUERTE                       False
ENTIDAD_FEDERATIVA           False
CODIGO_ENTIDAD_FEDERATIVA    False
ESTABLECIMIENTO              False
INSTITUCION                  False
EDAD_ANIOS                   False
FECHA_PROCURACION            False
RINON_IZQUIERDO              False
RINON_DERECHO                False
RINON_BLOCK                  False
PULMON_IZQUIERDO             False
PULMON_DERECHO               False
CORAZON                      False
HIGADO                       False
PANCREAS                     False
INTESTINO                    False
CORNEA_IZQUIERDA             False
CORNEA_DERECHA               False
PIEL                         False
HUESOS                       False
CORAZON_TEJIDOS              False
dtype: bool

De acuerdo con lo anterior, ahora sabemos que 2 columnas (SEXO, CODIGO_SEXO) tiene valores nulos, y utilizamos la siguiente función para saber exactamente cuantas celdas de estas columnas están vacías

In [19]:
df_org.isnull().sum()

SEXO                         4
CODIGO_SEXO                  4
TIPO_DONANTE                 0
MUERTE                       0
ENTIDAD_FEDERATIVA           0
CODIGO_ENTIDAD_FEDERATIVA    0
ESTABLECIMIENTO              0
INSTITUCION                  0
EDAD_ANIOS                   0
FECHA_PROCURACION            0
RINON_IZQUIERDO              0
RINON_DERECHO                0
RINON_BLOCK                  0
PULMON_IZQUIERDO             0
PULMON_DERECHO               0
CORAZON                      0
HIGADO                       0
PANCREAS                     0
INTESTINO                    0
CORNEA_IZQUIERDA             0
CORNEA_DERECHA               0
PIEL                         0
HUESOS                       0
CORAZON_TEJIDOS              0
dtype: int64

#### Cambiar nombres de columnas.  
Decidimos no realizar ningún cambio, ya que nos parece que tienen un nombre adecuado.

#### Modificaciones a la base de datos
Antes de generar las agrupaciones, es necesario eliminar los valores nulos de nuestra base, ya que no aportarán nada a nuestro análisis, además, al contar con pocas celdas vacías, no afectará nuestro objetivo quitar sus filas correspondientes.

In [22]:
df_org = df_org.dropna()

Una vez hecho lo anterior, es importante verificar que se haya realizado el cambio, por lo cual volvemos a mostrar la cantidad de celdas vacías que hay en nuestras columnas.

In [23]:
df_org.isnull().sum()

SEXO                         0
CODIGO_SEXO                  0
TIPO_DONANTE                 0
MUERTE                       0
ENTIDAD_FEDERATIVA           0
CODIGO_ENTIDAD_FEDERATIVA    0
ESTABLECIMIENTO              0
INSTITUCION                  0
EDAD_ANIOS                   0
FECHA_PROCURACION            0
RINON_IZQUIERDO              0
RINON_DERECHO                0
RINON_BLOCK                  0
PULMON_IZQUIERDO             0
PULMON_DERECHO               0
CORAZON                      0
HIGADO                       0
PANCREAS                     0
INTESTINO                    0
CORNEA_IZQUIERDA             0
CORNEA_DERECHA               0
PIEL                         0
HUESOS                       0
CORAZON_TEJIDOS              0
dtype: int64

Y finalmente, también eliminaremos las columnas de códigos, ya que consideramos que no son necesarios para nuestro fin

In [25]:
df_org = df_org.drop(['CODIGO_SEXO', 'CODIGO_ENTIDAD_FEDERATIVA'], axis=1)

Para verificar lo anterior, volvemos a mostrar las columnas de nuestra base de datos

In [26]:
df_org.columns

Index(['SEXO', 'TIPO_DONANTE', 'MUERTE', 'ENTIDAD_FEDERATIVA',
       'ESTABLECIMIENTO', 'INSTITUCION', 'EDAD_ANIOS', 'FECHA_PROCURACION',
       'RINON_IZQUIERDO', 'RINON_DERECHO', 'RINON_BLOCK', 'PULMON_IZQUIERDO',
       'PULMON_DERECHO', 'CORAZON', 'HIGADO', 'PANCREAS', 'INTESTINO',
       'CORNEA_IZQUIERDA', 'CORNEA_DERECHA', 'PIEL', 'HUESOS',
       'CORAZON_TEJIDOS'],
      dtype='object')

#### Guardar nuestra base de datos modificada

In [28]:
NewBase = df_org
NewBase.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DonarOrganos_NewBase.csv', index=False)

#### Guardar otros archivos con agrupaciones.
De acuerdo con los objetivos dados anteriormente, se requiere generar agrupaciones por entidad federativa, así como por tipo de donante, para posteriormente realizar el análisis correspondiente.  
  * Agrupar por entidad federativa

In [29]:
df_org.ENTIDAD_FEDERATIVA.unique()

array(['NUEVO LEÓN', 'CIUDAD DE MEXICO', 'PUEBLA', 'QUERÉTARO', 'MÉXICO',
       'SAN LUIS POTOSÍ', 'GUANAJUATO', 'TABASCO', 'VERACRUZ',
       'BAJA CALIFORNIA', 'DURANGO', 'AGUASCALIENTES', 'SONORA',
       'MICHOACÁN', 'COAHUILA', 'CHIHUAHUA', 'SINALOA', 'MORELOS',
       'ZACATECAS', 'COLIMA', 'BAJA CALIFORNIA SUR', 'JALISCO', 'OAXACA',
       'YUCATÁN', 'GUERRERO', 'TAMAULIPAS', 'QUINTANA ROO', 'NAYARIT',
       'TLAXCALA', 'HIDALGO', 'CHIAPAS', 'CAMPECHE'], dtype=object)

In [ ]:
# Agrupar por entidad federativa
NL = df_org[df_org['ENTIDAD_FEDERATIVA']=='NUEVO LEÓN']
NL.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_NuevoLeon.csv', index=False)
CdMex = df_org[df_org['ENTIDAD_FEDERATIVA']=='CIUDAD DE MEXICO']
CdMex.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_CdMexico.csv', index=False)
Puebla = df_org[df_org['ENTIDAD_FEDERATIVA']=='PUEBLA']
Puebla.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Puebla.csv', index=False)
Queretaro = df_org[df_org['ENTIDAD_FEDERATIVA']=='QUERÉTARO']
Queretaro.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Queretaro.csv', index=False)
Mexico = df_org[df_org['ENTIDAD_FEDERATIVA']=='MÉXICO']
Mexico.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Mexico.csv', index=False)
SLP = df_org[df_org['ENTIDAD_FEDERATIVA']=='SAN LUIS POTOSÍ']
SLP.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_SLP.csv', index=False)
Guanajuato = df_org[df_org['ENTIDAD_FEDERATIVA']=='GUANAJUATO']
Guanajuato.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Guanajuato.csv', index=False)
Tabasco = df_org[df_org['ENTIDAD_FEDERATIVA']=='TABASCO']
Tabasco.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Tabasco.csv', index=False)
Veracruz = df_org[df_org['ENTIDAD_FEDERATIVA']=='VERACRUZ']
Veracruz.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Veracruz.csv', index=False)
BajaC = df_org[df_org['ENTIDAD_FEDERATIVA']=='BAJA CALIFORNIA']
BajaC.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_BajaC.csv', index=False)
Durango = df_org[df_org['ENTIDAD_FEDERATIVA']=='DURANGO']
Durango.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Durango.csv', index=False)
Aguascalientes = df_org[df_org['ENTIDAD_FEDERATIVA']=='AGUASCALIENTES']
Aguascalientes.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Aguascalientes.csv', index=False)
Sonora = df_org[df_org['ENTIDAD_FEDERATIVA']=='SONORA']
Sonora.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Sonora.csv', index=False)
Michoacan = df_org[df_org['ENTIDAD_FEDERATIVA']=='MICHOACÁN']
Michoacan.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Michoacan.csv', index=False)
Coahuila = df_org[df_org['ENTIDAD_FEDERATIVA']=='COAHUILA']
Coahuila.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Coahuila.csv', index=False)
Chihuahua = df_org[df_org['ENTIDAD_FEDERATIVA']=='CHIHUAHUA']
Chihuahua.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Chihuahua.csv', index=False)
Sinaloa = df_org[df_org['ENTIDAD_FEDERATIVA']=='SINALOA']
Sinaloa.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Sinaloa.csv', index=False)
Morelos = df_org[df_org['ENTIDAD_FEDERATIVA']=='MORELOS']
Morelos.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Morelos.csv', index=False)
Zacatecas = df_org[df_org['ENTIDAD_FEDERATIVA']=='ZACATECAS']
Zacatecas.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Zacatecas.csv', index=False)
Colima = df_org[df_org['ENTIDAD_FEDERATIVA']=='COLIMA']
Colima.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Colima.csv', index=False)
BajaCS = df_org[df_org['ENTIDAD_FEDERATIVA']=='BAJA CALIFORNIA SUR']
BajaCS.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_BajaCS.csv', index=False)
Jalisco = df_org[df_org['ENTIDAD_FEDERATIVA']=='JALISCO']
Jalisco.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Jalisco.csv', index=False)
Oaxaca = df_org[df_org['ENTIDAD_FEDERATIVA']=='OAXACA']
Oaxaca.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Oaxaca.csv', index=False)
Yucatan = df_org[df_org['ENTIDAD_FEDERATIVA']=='YUCATÁN']
Yucatan.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Yucatan.csv', index=False)
Guerrero = df_org[df_org['ENTIDAD_FEDERATIVA']=='GUERRERO']
Guerrero.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Guerrero.csv', index=False)
Tamaulipas = df_org[df_org['ENTIDAD_FEDERATIVA']=='TAMAULIPAS']
Tamaulipas.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Tamaulipas.csv', index=False)
QuintanaRoo = df_org[df_org['ENTIDAD_FEDERATIVA']=='QUINTANA ROO']
QuintanaRoo.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_QuintanaRoo.csv', index=False)
Nayarit = df_org[df_org['ENTIDAD_FEDERATIVA']=='NAYARIT']
Nayarit.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Nayarit.csv', index=False)
Tlaxcala = df_org[df_org['ENTIDAD_FEDERATIVA']=='TLAXCALA']
Tlaxcala.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Tlaxcala.csv', index=False)
Hidalgo = df_org[df_org['ENTIDAD_FEDERATIVA']=='HIDALGO']
Hidalgo.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Hidalgo.csv', index=False)
Chiapas = df_org[df_org['ENTIDAD_FEDERATIVA']=='CHIAPAS']
Chiapas.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Chiapas.csv', index=False)
Campeche = df_org[df_org['ENTIDAD_FEDERATIVA']=='CAMPECHE']
Campeche.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Campeche.csv', index=False)


  * Agrupar por tipo de donante

In [32]:
df_org.TIPO_DONANTE.unique()

array(['CADÁVER', 'VIVO'], dtype=object)

In [ ]:
Cadaver = df_org[df_org['TIPO_DONANTE']=='CADÁVER']
Cadaver.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Cadaver.csv', index=False)
Vivo = df_org[df_org['TIPO_DONANTE']=='VIVO']
Vivo.to_csv('/Users/ManuelOrlando/Desktop/FACULTAD/Licenciatura en Actuaria/6to Semestre (7mo)/Mineria de Datos/DO_Vivo.csv', index=False)